### PR TITLE
Improving Lesson 1.3 by mentioning longest match while scanning tokens

### LINKED ISSUE #3539: Tutorial lesson 1.4 doesn't mention maximal-munch lexing
The exercise in question asks the user to identify why this grammar isn't ambiguous for the input `abb`:
```k
module EXERCISE4
  syntax Expr ::= "a" Expr "b"
                | "abb"
                | "b"
endmodule
```

For experienced K users, the answer is clearly that the lexer tokenises `abb` before the parser is ever called, but I don't see where we explain longest-match lexing in the tutorial. We should add a short section explaining this when we introduce lexing and parsing.

### PR DESCRIPTION
**Pull request by Robertorosmaninho**
Fixes #3539.

It makes more sense to me to add it to Lesson 1.3 instead of Lesson 1.4, as we're introducing Parsing in this lesson. But I'm open to suggestions to improve the text and where it should be! 😄

### CODE REVIEW

**Comment by radumereuta** [comment]
@dwightguth is more familiar with the K tutorial and native English speaker. I will let him give the final approval.

File: k-distribution/k-tutorial/1_basic/03_parsing/README.md [hunk 1]
```diff
@@ -88,6 +88,13 @@ interprets terms according to the grammar defined by the developer. It is
 automatically converted into an AST of that term, and then the `colorOf`
 function is evaluated using the function rules provided in the definition.
 
+You can ask yourself: How does K match the strings between the double quotes?
+The answer is that K uses Flex to generate a scanner for the grammar. Flex looks
+for the longest possible match of a regular expression in the input. If there
+are ambiguities between 2 or more regular expressions, it will pick the one with
+the max precedence token. You can learn more about how Flex matching works
```
Comment (dwightguth) [inline k-distribution/k-tutorial/1_basic/03_parsing/README.md]: "`prec` attribute", not "precedence token"

**Review by dwightguth — APPROVED** [review]
approved but please make one minor change

File: k-distribution/k-tutorial/1_basic/03_parsing/README.md (hunk 1 above)
Comment (Robertorosmaninho) [reply k-distribution/k-tutorial/1_basic/03_parsing/README.md]: Done!

**Pull request merged**